### PR TITLE
#1400. Tests for constructors of inline classes added

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good! Just one little thing:

The `@assertion` in LanguageFeatures/Inline-classes/static_analysis_member_invocation_A05_t03.dart doesn't match the topic of the test.

There is no text in the feature specification covering this specific topic. That's because it relies on a more general rule ("it's just like other class declarations"), and we will probably not be able to avoid that situation entirely.

I'm not quite sure how to handle this, because we're surely going to need it again. I'm suggesting that the `@assertion` is modified to say something relevant, and perhaps changed to use `@somethingElseThanAssertion` to indicate that it isn't from the spec.

WDYT?

File: LanguageFeatures/Inline-classes/static_analysis_member_invocation_A05_t03.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion If Dm is a method with function type F, and args exists, the
+/// static analysis of the inline member invocation is the same as that of an
+/// invocation with argument part args of a function with type [T1/X1 .. Ts/Xs]F
+///
+/// @description Checks static type of an inline class constructor tear-off
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Inline-classes/static_analysis_member_invocation_A05_t03.dart:10]: We do use `@note` for such cases. So I added `@note Inline classes specification doesn't contain this statement but assumes it`

[the remark this answers is not in the document]
Comment (eernstg) [reply LanguageFeatures/Inline-classes/static_analysis_member_invocation_A05_t03.dart:10]: Great, thanks!

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**